### ISSUE TITLE
[$250] Screen Readers: Many Pages: Locked switch buttons are not announced as locked or disabled

### ISSUE THREAD

**Issue by nbhardwaj-del** [Expensify/App#76957]
## Action Performed:
Prerequisite:
- User is logged in

1. Using Chrome+JAWS, open the URL: https://new.expensify.com/
2. Tab to focus on the required workspace and press Enter to activate it
3. Once the Workspace menu is displayed, tab to focus on the "Rules" menu item and press Enter to activate it
4. Tab to focus on the Switch button of eReceipts with a lock icon
5. Note how the control is announced.

Other Occurrences :
1. On Workspaces - Distance Rates
2. On Workspaces - Distance Rates - Rate Details

## Expected Result:
When a switch is visually locked or disabled, it should be programmatically identified as such. Screen readers should announce it as “eReceipts, switch, on, disabled” or “locked” to convey its state accurately.

## Actual Result:
The visually locked switch buttons (with a lock icon) are announced only as “eReceipts, switch, on” without conveying that the control is locked or disabled. This makes it unclear to screen reader users that the switch cannot be interacted with.

HTML:
<button aria-checked="false" aria-label="eReceipts" role="switch" data-tag="pressable" tabindex="0" class="css-g5y9jx r-1otgn73 r-1tazni7 r-17b9qp5" type="button" data-component="GenericPressable" data-element="Pressable" data-source-file="BaseGenericPressable.tsx"><div class="css-g5y9jx" id="419" style="width: 50px; height: 28px; justify-content: center; border-radius: 20px; padding: 15px; background-color: rgb(162, 169, 163);"><div class="css-g5y9jx" id="420" style="width: 22px; height: 22px; border-radius: 11px; position: absolute; left: 4px; justify-content: center; align-items: center; background-color: rgb(252, 251, 249); transform: translateX(0px);"><div class="css-g5y9jx" data-testid=""><svg xmlns="[http://www.w3.org/2000/svg"](http://www.w3.org/2000/svg%22) xml:space="preserve" viewBox="0 0 20 20" width="12" height="12" hovered="false" pressed="false" fill="#A2A9A3"><path d="M13.7 8V6c0-2.2-1.8-4-4-4s-4 1.8-4 4v2h-1c-.6 0-1 .4-1 1v6c0 1.7 1.3 3 3 3h6c1.7 0 3-1.3 3-3V9c0-.6-.4-1-1-1zm-2 0V6c0-1.1-.9-2-2-2s-2 .9-2 2v2z" style="fill-rule: evenodd; clip-rule: evenodd;"></path></svg></div></div></div></button>

## Area issue was found in:
Workspace Rules

## Failed WCAG checkpoints
WCAG 1.1.1 Non-Text Content
https://www.w3.org/WAI/WCAG22/Understanding/non-text-content.html

## User impact:
Users who have difficulty perceiving visual content use assistive technology to read text aloud, present it visually, or convert it to braille. Text alternatives may help some people who have difficulty understanding the meaning of photographs, drawings, and other images (e.g., line drawings, graphic designs, paintings, three-dimensional representations), graphs, charts, animations, etc. People who are deaf-blind can read the text alternatives in braille.

## Suggested resolution:
Use the aria-disabled="true" attribute to indicate that the switch is locked or not operable.
https://www.w3.org/WAI/ARIA/apg/patterns/switch/

An alternative is to use an off-screen text to announce the icon as 'locked'.

## Environment(s)
Windows Windows 11 - Chrome

## Screenshots/Videos
https://github.com/user-attachments/assets/15502b81-fc61-4a4f-88a7-691e39cb4f23

<img width="1919" height="971" alt="Image" src="https://github.com/user-attachments/assets/7ad88a63-3017-4878-b2e4-7022d04d4d76" />

**Version Number:** NA
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** NA
**Logs:** https://stackoverflow.com/c/expensify/questions/4856 NA
**Notes/Photos/Videos:** Any additional supporting documentation Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36605/community-issues/6996840
**Issue reported by:** NA
**Slack conversation:** NA

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~022014003668021804738</li>
        <li>Upwork Job ID: 2014003668021804738</li>
        <li>Last Price Increase: 2026-01-21</li>
<li>Automatic offers: </li>
<ul>
<li>abzokhattab | Reviewer | 110228799</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @mallenexpensify</details>

**Comment by MelvinBot** [comment]
Bug reproducible - https://platform.applause.com/services/links/v1/external/a014baf2a1a0d959de7b4539b0676976a4730626a5692088af025dacc699b431
https://platform.applause.com/services/links/v1/external/de36510efaf19030186d4f58ddae0dae76d4d78c544c084232db779cd6b52bc3

**Comment by rushatgabhane** [comment]
@nkdengineer ahhh great point! In that case we shouldn't need the PR

Thank youu @nkdengineer

**Comment by mallenexpensify** [comment]
Waiting on word back from QA

**Comment by bchapman-accessibilty** [comment]
@rushatgabhane It looks like "locked" was added to the label (Workflows > Auto-pay approved reports button). If this is the case for all locked buttons then the bug is fixed as it's currently written.

**Comment by rushatgabhane** [comment]
This issue was retested in **BFV cycle 544083 (Web)** on 2026-04-28 and is **Not Fixed**.

### Screenshots / Videos
- https://platform.applause.com/services/links/v1/external/8f3ade4f8e24ecb6040e0c9c75153a551d4aba8e17eea4f4600cda3ee509143c

### Retest notes
The issue is partially fixed

Main issue:
The UI has been changed, and the lock switch icon is not currently available.
- The toggle buttons are announced as 'Switch on/off' by the screen reader.

The other occurrence issues are not fixed:
- The locked state of the switch button is not announced by the screen reader.

**Comment by nkdengineer** [comment]
@rushatgabhane The video is the same as the previous case, and we discussed this [here](https://github.com/Expensify/App/issues/76957#issuecomment-4024473411).

**Comment by rushatgabhane** [comment]
cool, closing https://github.com/Expensify/App/issues/76957#issuecomment-4125181596
